configuration/config2/InsertableItem.hpp

```diff
@@ -0,0 +1,42 @@
+/*##############################################################################
+
+HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems�.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+############################################################################## */
+
+
+#ifndef _CONFIG2_CONFIGINSERTABLE_ITEMS_HPP_
+#define _CONFIG2_CONFIGINSERTABLE_ITEMS_HPP_
+
+#include "SchemaItem.hpp"
+#include <vector>
+
+struct InsertItemLimitChoice
```

Comment:
Do you use struct just to avoid a "public:" declaration?

Reply:
I suppose yes. It has been a style I have used that when a small simply all public class is necessary, I just use a struct to indicate that it's all public. I also use it when instantiated objects are treated more like data, such as items in a vector.